src/battle/action_cmd/mega_shock.c

```diff
@@ -1,13 +1,370 @@
+            if (actionCommandStatus->hudElementX > 50) {
+                actionCommandStatus->hudElementX = 50;
+            }
+
+            set_hud_element_render_pos(actionCommandStatus->hudElements[0], actionCommandStatus->hudElementX - 17,
+                actionCommandStatus->hudElementY);
+            set_hud_element_render_pos(actionCommandStatus->hudElements[2], actionCommandStatus->hudElementX + 23,
+                actionCommandStatus->hudElementY);
+            set_hud_element_render_pos(actionCommandStatus->hudElements[1], actionCommandStatus->hudElementX,
+                actionCommandStatus->hudElementY + 28);
+            return;
+        case 10:
+            btl_set_popup_duration(99);
+            if (actionCommandStatus->unk_4E != 0) {
+                actionCommandStatus->unk_4E--;
+                return;
+            }
+            set_hud_element_anim(actionCommandStatus->hudElements[0], &D_80108B80);
+            set_hud_element_anim(actionCommandStatus->hudElements[2], &D_80292374);
+            actionCommandStatus->barFillLevel = 0;
+            actionCommandStatus->unk_5C = 0;
+            actionCommandStatus->unk_54 = actionCommandStatus->unk_52;
+            sfx_play_sound_with_params(0x80000041, 0, 0, 0);
+            actionCommandStatus->state = 11;
+        case 11:
+            btl_set_popup_duration(99);
+            if (actionCommandStatus->unk_68 == 0) {
+                s16 newFillLevel;
+
+                if (actionCommandStatus->unk_64 != 0) {
+                    s8 mashMeterIntervals = actionCommandStatus->mashMeterIntervals;
+                    s16* mashMeterCutoffs = actionCommandStatus->mashMeterCutoffs;
+                    s32 index;
+
+                    mashMeterCutoff = mashMeterCutoffs[mashMeterIntervals];
+                    index = actionCommandStatus->barFillLevel / mashMeterCutoff / 20;
+                    newFillLevel = actionCommandStatus->barFillLevel - D_802A9930_42E340[index].unk_8;
+                } else {
+                    newFillLevel = actionCommandStatus->barFillLevel - 10;
+                }
+
+                actionCommandStatus->barFillLevel = newFillLevel;
+                if (actionCommandStatus->barFillLevel << 10 < 0) {
+                    actionCommandStatus->barFillLevel = 0;
+                }
+            }
+
+            bufferPos = battleStatus->inputBufferPos;
+            bufferPos -= 2;
+            if (bufferPos < 0) {
+                bufferPos += ARRAY_COUNT(battleStatus->pushInputBuffer);
+            }
+
+            buttonsPushed = 0;
+            frameCount = 1;
+
+            while(frameCount >= 0) {
+                if (bufferPos >= ARRAY_COUNT(battleStatus->pushInputBuffer)) {
+                    bufferPos -= ARRAY_COUNT(battleStatus->pushInputBuffer);
+                }
+
+                buttonsPushed |= battleStatus->pushInputBuffer[bufferPos++];
+                frameCount--;
+            }
+
+            buttonsAB = BUTTON_A | BUTTON_B;
```

Comment:
Is this temp needed? I'd imagine you can re-use A | B

Reply:
This needs to be in a temp or it causes a codegen difference, though it might be possible to rewrite the expression to not have this problem...